modules/topicsFpdModule.js

```diff
@@ -73,6 +100,54 @@ export function processFpd(config, {global}, {data = topicsData} = {}) {
   });
 }
 
+// Added function addListenerToFetchTopics to listen for topic data retured from iframe loaded for SSPs(bidder)
+function addListenerToFetchTopics() {
+  window.addEventListener('message', (e) => {
+    if (e && e.data) {
+      try {
+        let data = JSON.parse(e.data);
+        if (data && data.segment && !isEmpty(data.segment.topics)) {
+          let {domain, topics, bidder} = data.segment;
+          let segmentData = getTopicsData(domain, topics);
+          let updateStore = [];
+          let bidderFoundInStore = false;
+          let storedSegments = JSON.parse(storage.getDataFromLocalStorage(`${initialTopicName}${bidder}`));
+          storedSegments?.forEach((segment) => {
+            if (segment.name === segmentData[0].name && segment.ext.segclass == segmentData[0].ext.segclass) {
```

Comment:
This means that topics can never be removed from the list. I have no opinion on whether they should, but it does seem to me that this is working around the purpose of the topics API (limiting how much privacy sensitive data is moved around).

If we want to keep each topic forever, I do think we should use a map to store them here - looping through all of them can get quite expensive since the list size can only increase.